pkg/app/app.go

```diff
@@ -65,41 +65,40 @@ func ValidateFlags(bundle string, args []string, cmd *cobra.Command, opt *kobjec
 	logrus.Debug("Checking validation of provider %s", provider)
 
 	// OpenShift specific flags
-	deploymentConfig := cmd.Flags().Lookup("deployment-config").Changed
 	buildRepo := cmd.Flags().Lookup("build-repo").Changed
 	buildBranch := cmd.Flags().Lookup("build-branch").Changed
 
 	// Kubernetes specific flags
 	chart := cmd.Flags().Lookup("chart").Changed
-	daemonSet := cmd.Flags().Lookup("daemon-set").Changed
-	replicationController := cmd.Flags().Lookup("replication-controller").Changed
-	deployment := cmd.Flags().Lookup("deployment").Changed
 
 	// Check validations against provider flags
 	switch {
 	case provider == "openshift":
 		if chart {
 			logrus.Fatalf("--chart, -c is a Kubernetes only flag")
 		}
-		if daemonSet {
-			logrus.Fatalf("--daemon-set is a Kubernetes only flag")
+		if opt.CreateDS {
+			logrus.Fatalf("DaemonSet is a Kubernetes only controller")
```

Comment:
This value is coming from two places from commandline and from the preference file so changing the message to more generic form.